hypertension/dc-7101-algorithms/py-root/lib/algorithms/bp_sufficiency.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from .utils import tally_diastolic_counts, tally_systolic_counts, calculate_reading_from_bucket, bp_readings_meet_date_specs
+
+def sufficient_to_autopopulate (request):
+    """
+    Determine if there is enough BP data to calculate a predominant reading,
+    and if so return the predominant rating
+
+    :param request: request body
+    :type request: dict
+    :return: response body indicating success or failure with additional attributes
+    :rtype: dict
+    """
+
+    predominance_calculation = {
+        "success": True,
+    }
+    date_of_claim = request["date_of_claim"]
+    filter_bp_readings = []
+    date_of_claim_date = datetime.strptime(date_of_claim, "%Y-%m-%d").date()
+
+    if len(request["bp"]) < 1:
+        predominance_calculation["success"] = False
+        return predominance_calculation
+
+    for reading in request["bp"]:
+        bp_reading_date = datetime.strptime(reading["date"], "%Y-%m-%d").date()
+        if bp_reading_date >= date_of_claim_date - relativedelta(years=1):
```

Comment:
should we include the same day a year ago, or not? EX: the claim date is 12/1/2021, do we allow 12/1/2020? Depending on datetime (which not using), it could be included, but it seems that just by days, we wouldn't include it, right?

Reply:
going to write another jira ticket - resolved

Reply:
MCP-1045 - ABD: Ask VA SME's about including same date from previous year of look up date

Reply:
https://amida.atlassian.net/browse/MCP-1045

Reply:
(Resolved)